carma_wm/src/CARMAWorldModel.cpp

```diff
@@ -1425,21 +1476,22 @@ namespace carma_wm
         
         if (!curr_light->recorded_time_stamps.empty())
         {
-          boost::posix_time::time_duration time_difference = curr_light->predictState(min_end_time).get().first - min_end_time;
+          boost::posix_time::time_duration time_difference = curr_light->predictState(min_end_time - lanelet::time::durationFromSec(0.5)).get().first - min_end_time; //0.5s to account for error
           ROS_DEBUG_STREAM("Initial time_difference: " << (double)(time_difference.total_milliseconds() / 1000.0));
-          if (curr_light->predictState(min_end_time).get().second !=  received_state)
+          
+          if (curr_light->predictState(min_end_time - lanelet::time::durationFromSec(0.5)).get().second !=  received_state)
```

Comment:
ditto here why 0.5s?

Reply:
SPAT timestamps can drift for 0.1s, so the error covers that. 
but also timetamp may fall exactly on min_end_time and pick the new signal state, thinking SPAT changed. So 0.5s also makes sure we don't do that. Smaller number like 0.3s may have worked, but haven't tested whereas 0.5s works reliably now.